### PR TITLE
feat: add monaco adaptor to component explorer site

### PR DESCRIPTION
**Pull request by janechu**
# Description

This change allows 2 way editing via the new `MonacoAdaptor` in the Component explorer.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

**Adding or modifying component(s) in `@microsoft/fast-components` checklist**

- [ ] I have added a new component
- [ ] I have modified an existing component
- [ ] I have updated the [definition file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#definition)
- [ ] I have updated the [configuration file](https://github.com/Microsoft/fast/blob/master/packages/web-components/fast-components/CONTRIBUTING.md#configuration)

## Process & policy checklist

- [ ] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://www.fast.design/docs/community/code-of-conduct/#our-standards) for this project.

### CODE REVIEW

File: sites/fast-component-explorer/app/explorer.tsx [hunk 1]
```diff
@@ -118,6 +121,10 @@ class Explorer extends Foundation<
     private windowResizing: number;
     private editor: monaco.editor.IStandaloneCodeEditor;
     private editorContainerRef: React.RefObject<HTMLDivElement> = React.createRef();
+    private adaptor: MonacoAdaptor;
```
Comment (EisenbergEffect) [inline sites/fast-component-explorer/app/explorer.tsx:124]: "adapter"

**Review by EisenbergEffect — APPROVED** [review]

File: sites/fast-component-explorer/app/explorer.tsx (hunk 1 above)
Comment (janechu) [reply sites/fast-component-explorer/app/explorer.tsx:124]: Well, I am going to have to do a scrub after this haha, good thing I didn't export it from root.

**Comment by janechu** [comment]
There is an outstanding issue with regards to updated `dictionaryId` which is still being resolved.

**Review by scomea — APPROVED** [review]

**Comment by janechu** [comment]
Placing this on hold while investigating two additional issues:
1. Data navigation issue (PR issued: https://github.com/microsoft/fast/pull/3955)
2. Performance issue where changes should be aggregated before executing due to Monaco initialization taking some time
3. Additional newlines and other formatting being recorded as text in the DataDictionary which should be removed prior to conversion

File: sites/fast-component-explorer/app/explorer.tsx (hunk 1 above)
Comment (janechu) [reply sites/fast-component-explorer/app/explorer.tsx:127]: Issue filed to fix this - https://github.com/microsoft/fast/issues/3961

At this time the MonacoAdaptor has not been exported from root or documented so this is a safe non-breaking change.

**Review by chrisdholt — APPROVED** [review]

**Pull request merged**